### PR TITLE
#118 Implement optimization methods 

### PR DESCRIPTION
**Pull request by jrozal**

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -5,6 +5,13 @@ const cors = require('cors');
 const db = require('../database/database.js');
 
 app.use(cors());
+
+app.get('*.js', function (req, res, next) {
```
Comment (amosNash) [inline server/server.js:9]: Can you explain what this get request is doing? Why are you passing in '*.js' as a parameter?

File: client/components/App.jsx [hunk 2]
```diff
@@ -66,7 +73,7 @@ const CustomerQuestions = () => {
 
 
   const getCustomerQuestionsData = (productId) => {
-    axios.get(`http://ec2-3-22-93-125.us-east-2.compute.amazonaws.com:4001/customer-questions/${productId}`)
+    axios.get(`http://localhost:4001/customer-questions/${productId}`)
```
Comment (amosNash) [inline client/components/App.jsx]: Any reason for switching to making a request to your local host?

File: client/components/App.jsx [hunk 3]
```diff
@@ -66,7 +73,7 @@ const CustomerQuestions = () => {
...
       .then(res => {
         console.log(res);
```
Comment (amosNash) [inline client/components/App.jsx]: Can get rid of console.log here.

File: webpack.config.js [hunk 4]
```diff
@@ -27,5 +28,14 @@ module.exports = {
         ],
       },
     ]
-  }
+  },
+  plugins: [
+    new CompressionPlugin({
```
Comment (amosNash) [inline webpack.config.js:33]: Can you explain what this plugin is doing? Is it compressing your files?

**Review by amosNash — COMMENTED** [review]
Looks good, I didn't have many comments for this PR.

File: server/server.js (hunk 1 above)
Comment (jrozal) [reply server/server.js:9]: Sure, this code allows the express server to serve the zipped (optimized) version of my bundle to the client.
This stackoverflow post goes into it a bit: https://stackoverflow.com/questions/49911058/cant-serve-gzipped-files-properly-using-express
I would consider looking into it! It turned some of my services from ratings of 70 to 100!

File: client/components/App.jsx (hunk 2 above)
Comment (jrozal) [reply client/components/App.jsx]: Sorry, this was committed by accident. I haven't set up environment variables and I am still working on features!

File: webpack.config.js (hunk 4 above)
Comment (jrozal) [reply webpack.config.js:33]: Yep! I am using the compression webpack plugin to compress my bundle file. https://www.npmjs.com/package/compression-webpack-plugin

**Pull request merged**